File: swift/circle.swift
import Foundation

struct Circle {
    var radius: Double                             // stored property
    var area: Double {                             // computed property
        get {return Double.pi * radius * radius}
        set(a) {radius = sqrt(a/Double.pi)}
    }
}

var c = Circle(radius: 10)
assert(c.radius == 10 && c.area == 100*Double.pi)  // invokes get
c.area = 16*Double.pi                              // invokes set
assert(c.radius == 4 && c.area == 16*Double.pi)

File: rust/src/bin/circle.rs
use std::f64::consts::PI;
use std::fmt;

struct Circle {
    x: f64,
    y: f64,
    radius: f64,
}

impl Circle {
    fn area(&self) -> f64 {
        PI * (self.radius * self.radius)
    }

    fn circumference(&self) -> f64 {
        PI * 2f64 * self.radius
    }
}

impl fmt::Display for Circle {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(
            f,
            "Circle {{ x: {}, y: {}, radius: {} }}",
            self.x, self.y, self.radius
        )
    }
}

fn main() {
    let origin = Circle { x: 1.0, y: 5.0, radius: 1.0};
    println!("{}", origin);
    println!("{} {}", origin.area(), origin.circumference());
}
